lib/bolt/project.rb

```diff
@@ -68,6 +87,15 @@ def plans
     end
 
     def validate
+      n = @data['name']
+      if n && !Puppet::Module.is_module_directory_name?(n) && !Puppet::Module.is_module_namespaced_name?(n)
+        raise Bolt::ValidationError, <<~ERROR_STRING
+        Invalid module name '#{n}'; module names must match either:
```

Comment:
This error message isn't quite right for this use case. It shouldn't refer to "module" anywhere ("project" instead), and shouldn't refer to "installed" anything.